guides/source/developers/getting-started/customer-flow.html.md

```diff
@@ -0,0 +1,247 @@
+  - It checks whether the promotion is past its expiration date.
+  - It checks whether the promotion usage is exceeded.
+  - Only if the above checks pass is a `Spree::OrderPromotion` object
+    created.
+- The new `Spree::OrderPromotion` object associates the `Spree::Order` ID
+  with the `Spree::Promotion` ID.
+  - If the promotion only applies to a specific product (or multiple
+    products), that product's `Spree::LineItem` object is updated with a new
+   `promo_total` value.
+  - If the promotion only applies to a specific shipment (or multiple
+    shipments), that shipment's `Spree::Shipment` object is updated with a new
+    `promo_total` value.
+  - If the promotion applies to the entire order and now a specific line
+    item or shipment, only the `Spree::Order`'s	 `promo_total` value is
+    updated.
+- The `Spree::Order` object is updated.
+  - The `promo_total` is no longer `nil`. It is updated with the values
+    from all `promo_total` values associated with the order.
+
+[promotion-rules]: ../promotions/promotion-rules.html
+[spree-promotion]: ../promotions/overview.html
+
+Customer is prompted to enter payment information. 
+
+- The customer chooses an option from the listed payment methods
+  ([`Spree::PaymentMethod`s][spree-payment-method]).
+  - If customer has placed an order previously and saved payment
+    information, the customer can use the payment details on file or
+    choose a different payment method.
+  - A new customer can choose one of the listed `Spree::PaymentMethod`s.
+- Depending on which payment method the customer selected, the payment method
+  uses its associated [payment service provider][payment-service-providers] that
+  will process the payment.
+- Based on the payment method, a [payment source][spree-payment-source] can be
+  determined.
+- A new [`Spree::Payment`][spree-payment] object that uses a `order_id` to
+  associate with the `Spree::Order`.
+  - The payment is associated with the `Spree::PaymentMethod`'s ID.
+  - The payment has a `source_type` and `source_id` associated with the
+    `Spree::PaymentMethod` and payment source being used.
+  - The `state` of the payment is `checkout`.
+- The `Spree::Order` object is updated.
+  - The `payment_state` is no longer `nil` and is now `balance_due`.
+
+[payment-service-providers]: ../payments/payment-service-providers.html
+[spree-payment]: ../payments/payments.html
+[spree-payment-method]: ../payments/payment-methods.html 
+[spree-payment-source]: ../payments/payment-sources.html
+
+Customer enters their preferred payment details and submits them.
+
+Customer reviews the confirmation screen and places order.
+
+- The `Spree::Payment`'s `state` changes from `checkout` to `pending`.
+
+Administrator approves the shipment.
+
+- The `Spree::Order`'s `approver_id` and `approved_at` values are no
+  longer `nil`.
+- The [`Spree::StockItem`'s][spree-stock-item] `count_on_hand` value decrease by
```

Comment:
doesn't this happen when the shipment is shipped?

Reply:
I will run through an order and double-check this now. Thank you.

My understanding was that a sold stock item is no longer counted as on hand, because it has been sold and will be shipped.